### ISSUE TITLE
[v5] input fields of type number not properly supporting negative values?

### ISSUE THREAD

**Issue by mdonthireddy** [erikras/redux-form#1470]
It is quite possible I am missing something simple as I am relatively inexperienced in web and Javascript development. But this seems like it could be a genuine problem worth reporting.

The issue : I am unable to type a minus sign into an empty input of type number. It seems like the issue is that Redux-form's onChange handler tries to parse the field value to a float. When the only character is a "-", parseFloat() returns NaN, and the ultimate result is that "-" is replaced with an empty string in the input field.

WORKAROUND: 
1) When I need a negative number such as -2, I type the "2" first and insert the "-" later.
2) After some poking around, I found that editing node_modules/redux-form/lib/events/getValue.js and removing the parseFloat() call seems to work:
`if (type === 'number' || type === 'range') {
              // parseFloat(value);
              return value;
    }
`

**Comment by mdonthireddy** [comment]
BTW I am using redux-form version 5.3.1

Also, my code at the end of the post got reformatted without newlines. What I meant to say is that I replaced "parseFloat(value)" with just "value".

**Comment by oakley808** [comment]
Same appears to be true for v6.0.2. You have to hit the minus sign twice to get it to insert first. (or type in the number first and go back and add the minus sign.)

**Comment by oakley808** [comment]
Any movement on this? I think the relevant code in 6.x is
https://github.com/erikras/redux-form/blob/e9bcc1b54e211071f0cefc644c1ab7b2e93b9d78/src/events/getValue.js#L34-L36

**Comment by shanSyapse** [comment]
There is a related issue that is closed: https://github.com/erikras/redux-form/issues/1383
However, for input type === 'number', if user type in '-', e.targe.value === ''; 
saving input value as empty string overwrite '-', that is why the '-' can not show up.

So that's why this issue still exists.

**Comment by 3ffy** [comment]
Still get the same issue in redux-form v 6.4.3 . 

I understand that parseFloat is important at some point to internally validating numbers, but i think a simple bugfix would be something like:

```javascript
value = (value === '+' || value === '-') ? value : parseFloat(value);
```

This way the input form will keep single char operator for the second char, and then only check if it is a number. It looks really less strange for the user because caracters won't disapear anymore.

PS: sorry i'm pretty new to react & git, i can't do a PR by myself :').

**Comment by Rafkraft** [comment]
I resolved this buy rendering `<input type="text">` instead of `<input type="number">`, even for float values.
This way the user is free and can type a negative number freely without the minus bug. Therefore I added a validation rule to make sure that the input is correct (float in this occurrence):
```
  if (parseFloat(values.fieldName).toString().length !== values.fieldName.length) {
    errors.fieldName = 'Invalid value for fieldName';
  }
```

In the `onSubmit` method passed to redux-form, simply `parseFloat` or `parseInt` the value.

Of course, the arrows to increment/decrement the number wont be here, and the user is free to enter alpha and special characters in the `input` but it wont pass the validate step, and in my case, I consider the user to be mature enough to know what he does.
